core/Object.php

```diff
@@ -213,6 +213,9 @@ public static function create_from_string($classSpec, $firstArg = null) {
 	 * Returns a 2-elemnent array, with classname and arguments
 	 */
 	public static function parse_class_spec($classSpec) {
+		if ($classSpec === array()) {
```

Comment:
This is a little obscure, where the underlying reason is "sometimes code passes in an empty array in some edge case". It'd be better to never call this method with an invalid arg, rather than silently ignoring the error.